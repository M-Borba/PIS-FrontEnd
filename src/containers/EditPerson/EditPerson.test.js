import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import ReactDOM from "react-dom";
import Edit from "./index";
import React from "react";

describe("project list component tests", () => {
  it("renders without crashing", () => {
    const pData = {
      first_name: "A",
      last_name: "B",
      email: "A@effectus.com",
      working_hours: 40,
    };
    const div = document.createElement("div");
    ReactDOM.render(<Edit personData={pData} id={1} true />, div);
  });
  it("renders correct data", () => {
    const pData = {
      first_name: "A",
      last_name: "B",
      email: "A@effectus.com",
      working_hours: 40,
    };
    const { container } = render(<Edit personData={pData} id={1} true />);
    // Matches the last snapshot taken
    expect(container).toMatchSnapshot();
  });
});
