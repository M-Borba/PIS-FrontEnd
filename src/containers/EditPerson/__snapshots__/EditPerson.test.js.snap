// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`project list component tests renders correct data 1`] = `
<div>
  <div>
    <div
      class="makeStyles-paper-1"
    >
      <h1
        class="MuiTypography-root MuiTypography-h5"
      >
        Editando Persona
      </h1>
      <h2
        class="MuiTypography-root makeStyles-msg-5 MuiTypography-body1"
      />
      <form
        class="makeStyles-form-2"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
            data-shrink="true"
            for="first_name"
            id="first_name-label"
          >
            Nombre
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              id="first_name"
              name="first_name"
              required=""
              type="text"
              value="A"
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-8 PrivateNotchedOutline-legendNotched-9"
              >
                <span>
                  Nombre
                   *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
            data-shrink="true"
            for="last_name"
            id="last_name-label"
          >
            Apellidos
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              id="last_name"
              name="last_name"
              required=""
              type="text"
              value="B"
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-8 PrivateNotchedOutline-legendNotched-9"
              >
                <span>
                  Apellidos
                   *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-focused Mui-focused Mui-required Mui-required"
            data-shrink="true"
            for="email"
            id="email-label"
          >
            Email Address
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              autocomplete="email"
              class="MuiInputBase-input MuiOutlinedInput-input"
              id="email"
              name="email"
              required=""
              type="email"
              value="A@effectus.com"
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-8 PrivateNotchedOutline-legendNotched-9"
              >
                <span>
                  Email Address
                   *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
            data-shrink="true"
            for="working_hours"
            id="working_hours-label"
          >
            Horas
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              id="working_hours"
              name="working_hours"
              required=""
              type="number"
              value="40"
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-8 PrivateNotchedOutline-legendNotched-9"
              >
                <span>
                  Horas
                   *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-3 MuiButton-containedPrimary MuiButton-fullWidth"
          role="submit"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Guardar
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <h2
          class="MuiTypography-root makeStyles-errorMsg-4 MuiTypography-body1"
        />
        <div
          class="MuiBox-root MuiBox-root-11"
        />
      </form>
    </div>
  </div>
</div>
`;
